docs/authentication.md

```diff
@@ -0,0 +1,211 @@
+# Authentication Functions
+
+<!-- Glee relies on functions to execute your business logic. Functions are files that export a default async Node.js function: -->
+
+Authentication in Glee can be done using authentication functions. Authentication functions are files that export either one or both of the `clientAuth` and `serverAuth` Node.js functions:
+
+```js
+/* websocket.js */
+
+export async function serverAuth({ authProps, done }) {
+  //server auth logic
+}
+
+export async function clientAuth({ parsedAsyncAPI, serverName }) {
+  //client auth logic
+}
+```
+
+Glee looks for authentication files in the `auth` directory, the name of the authentication file should be the name of the targeted server that the authentication logic should work for.
+
+## Supported Authentication Values in AsyncAPI.yaml file
```

Comment:
Instread of providing an example here I think it would be great to have a few other documents providing a getting started guide for that type of authentication?
maybe:
`auth/user-pass.md` explaining the user-pass authentication using a user name and password. it can be for `http` `or websocket`.
`auth/oauth.md` would do the same but for the oauth flow...
etc...